src/app/json_test.cpp

```diff
@@ -85,7 +85,7 @@ TEST_CASE("active-operational-dataset-encoding-decoding", "[json]")
 
         std::string              json = ActiveDatasetToJson(dataset);
         ActiveOperationalDataset dataset1;
-        REQUIRE(ActiveDatasetFromJson(dataset1, json) == Error::kNone);
+        REQUIRE(ActiveDatasetFromJson(dataset1, json).IsNone());
```

Comment:
I feel this is not helping with readability of the code and that we are checking the there is no error.

I would suggest the following: In `Error` class we define an overload of `operator==(ErrorCode aCode)` which compares an `Error` with a `ErrorCode` enum value. This would allow original code (which in my opinion is more readable) to continue to work.

Reply:
Agree that `IsNone()` is confusing in some situations.

Reply:
I used to suggest `IsSuccess()` for checking if the operation (error) was successful (none):

- `ActiveDatasetFromJson(dataset1, json).IsSuccess()` equally readable as (if not more readable than)  `ActiveDatasetFromJson(dataset1, json) == Error:kNone`
- `.IsSuccess()` is simply more consise than ` == Error::kNone`
- `.IsSuccess()` works better with editor's `autocomplete` feature (most of time we just need to type in `.`).

Thoughts? @wgtdkp @abtink @jwhui

Reply:
It will be a little weird when we defined a error:
```C++
Error error;
...
exit:
    if (error.IsSuccess())
    {
        ...
    }
```

`error is success` reads strange.

Reply:
I agree `error.IsSuccess()` may not be idle, but personally I think it is ok and would prefer `IsSuccess()` over `IsNone()` for statements like: `ActiveDatasetFromJson(dataset1, json).IsSuccess()`

Reply:
@simonlingoogle @abtink Thanks for your advices! I think `operator==` works good in both situations. I have revised to use `==`, please review.